sys/net/gnrc/network_layer/ipv6/nib/_nib-router.c

```diff
@@ -190,7 +193,11 @@ void _set_rtr_adv(gnrc_netif_t *netif)
 static void _snd_ra(gnrc_netif_t *netif, const ipv6_addr_t *dst,
                     bool final, _nib_abr_entry_t *abr)
 {
-    gnrc_pktsnip_t *ext_opts = _build_ext_opts(netif, abr);
+    gnrc_pktsnip_t *ext_opts = NULL;
+
+    if (final) {
+        _build_ext_opts(netif, abr);
```

Comment:
Shouldn't this be `ext_opts = _build_ext_opts(netif, abr);`?

Reply:
Should have tested the negative case as well ;-).